python/tree_reader.py

```diff
@@ -94,28 +97,40 @@ def getDataFrame(self, prefix, entry_block, fallback=None):
                     if br.startswith(f'{prefix}_') and
                     br != f'{prefix}_n']
         names = ['_'.join(br.split('_')[1:]) for br in branches]
-        name_map = dict(zip(names, branches, strict=False))
+        name_map = dict(zip(names, branches))
         if len(branches) == 0:
             if fallback is not None:
                 return self.getDataFrame(prefix=fallback, entry_block=entry_block)
             prefs = set([br.split('_')[0] for br in self._branches])
             print(f'stored branch prefixes are: {prefs}')
             raise ValueError(f'[TreeReader::getDataFrame] No branches with prefix: {prefix}')
 
-        akarray = self.tree.arrays(names,
-                                   library='ak',
-                                   aliases=name_map,
-                                   entry_start=self.file_entry,
-                                   entry_stop=self.file_entry+entry_block)
+        dask_akarray = NanoEventsFactory.from_root(
```

Comment:
is this still up to date? did you try avoiding the loopo over the files opening all of them at once moving the `NanoEventsFactory` to the `analyzer` module?

Reply:
Now it is up-to-date.